File: csharp/ToolGood.Algorithm/Internals/Functions/Value/Function_NULL.cs
using System;
using System.Collections.Generic;
using System.Text;
using ToolGood.Algorithm.Enums;
using ToolGood.Algorithm.Internals;

namespace ToolGood.Algorithm.Internals.Functions.Value
{
	internal sealed class Function_NULL : Function_0
	{
		public Function_NULL()
		{
		}

		public override string Name => "NULL";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter = null)
		{
			return Operand.Null;
		}

		public override OperandType GetResultType()
		{
			return OperandType.NULL;
		}

		public override void ToString(StringBuilder stringBuilder, bool addBrackets)
		{
			stringBuilder.Append("NULL");
		}

	}
}

File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/internals/functions/value/Function_NULL.java
package toolgood.algorithm.internals.functions.value;

import java.util.function.BiFunction;
import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.functions.Function_0;

public final class Function_NULL extends Function_0 {

    public Function_NULL() {
    }

    @Override
    public String Name() {
        return "NULL";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        return Operand.Null;
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.NULL;
    }

    @Override
    public void ToString(StringBuilder stringBuilder, boolean addBrackets) {
        stringBuilder.append("NULL");
    }
}
